tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,171 @@
+from tap_tester import runner, connections, menagerie
+from base import AsanaBase
+from dateutil import parser as parser
+
+
+class AsanaInterruptedSyncTest(AsanaBase):
+    """
+    Test to verify that if a sync is interrupted, then the next sync will continue
+    from the bookmarks and currently syncing stream.
+    """
+
+    def name(self):
+        return "tap_tester_asana_interrupted_sync_test"
+
+    def test_run(self):
+        """
+        Scenario: A sync job is interrupted. The state is saved with `currently_sync_stream`.
+                  The next sync job kicks off and the tap picks back up on that
+                  `currently_sync_stream` stream.
+
+        Expected State Structure:
+            {
+                "currently_sync_stream": "stream_name",
+                "bookmarks": {
+                    "stream_1": {"bookmark_field": "2010-10-10T10:10:10.100000"},
+                    "stream_2": {"bookmark_field": "2010-10-10T10:10:10.100000"}
+                }
+            }
+
+        Test Cases:
+        - Verify an interrupted sync can resume based on the `currently_sync_stream` and
+            stream level bookmark value.
+        - Verify only records with replication-key values greater than or equal to the
+            stream level bookmark are replicated on the resuming sync for the interrupted stream.
+        - Verify the yet-to-be-synced streams are replicated following the interrupted stream
+            in the resuming sync.
+        """
+
+        self.start_date = self.get_properties()["start_date"]
+        start_date_datetime = self.dt_to_ts(self.start_date, self.START_DATE_FORMAT)
+
+        conn_id = connections.ensure_connection(self)
+        expected_streams = {"projects", "tags", "tasks"}
+
+        # Run check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Select only the expected streams tables
+        catalog_entries = [ce for ce in found_catalogs if ce["tap_stream_id"] in expected_streams]
+
+        # Catalog selection
+        self.select_found_catalogs(conn_id, catalog_entries)
+
+        # Run a sync job
+        self.run_and_verify_sync(conn_id)
+        synced_records_full_sync = runner.get_records_from_target_output()
+
+        full_sync_state = menagerie.get_state(conn_id)
+
+        # State to run 2nd sync
+        #   projects: synced records successfully
+        #   tags: remaining to sync
+        #   tasks: currently getting synced
+        state = {
+            "bookmarks": {
+                "currently_sync_stream": "tasks",
+                "projects": {
+                    "modified_at": "2022-10-06T10:08:17.492000"
+                }
+            }
+        }
```

Comment:
```suggestion
        interrupted_sync_state = {
            "bookmarks": {
                "currently_sync_stream": "tasks",
                "projects": {
                    "modified_at": "2022-10-06T10:08:17.492000"
                }
            }
        }
```
Name the variable appropriately.

Reply:
In interrupted sync state ideally we should have bookmark greater than start date and less than current date.

Reply:
Updating intermittent interrupted bookmark for the interrupted stream.